zerver/lib/message.py

```diff
@@ -227,7 +228,10 @@ def message_to_dict_json(message: Message, realm_id: Optional[int] = None) -> by
 
 
 def save_message_rendered_content(message: Message, content: str) -> str:
-    rendered_content = render_markdown(message, content, realm=message.get_realm())
+    rendering_result = render_markdown(message, content, realm=message.get_realm())
+    rendered_content = None
+    if rendering_result is not None:
+        rendered_content = rendering_result.rendered_content
```

Comment:
It looks like `render_markdown` is annotated as always returning a `MessageRenderingResult`, so I think this can just be like the realm/stream description files.

Reply:
This is the place where we have a strange test called `test_applying_markdown_invalid_format` that patches `render_markdown` as mentioned earlier. But I'm reluctant to delete it because I'm not sure what it might break.

Reply:
OK.